vortex-zigzag/src/compress.rs

```diff
@@ -30,87 +27,119 @@ impl EncodingCompression for ZigZagEncoding {
         // Only compress if the array has negative values
         // TODO(ngates): also check that Stat::Max is less than half the max value of the type
         parray
-            .stats()
-            .get_or_compute_cast::<i64>(&Stat::Min)
+            .statistics()
+            .compute_as_cast::<i64>(Stat::Min)
             .filter(|&min| min < 0)
             .map(|_| self as &dyn EncodingCompression)
     }
 
     fn compress(
         &self,
-        array: &dyn Array,
-        like: Option<&dyn Array>,
+        array: &Array,
+        like: Option<&Array>,
         ctx: CompressCtx,
-    ) -> VortexResult<ArrayRef> {
-        let zigzag_like = like.map(|like_arr| like_arr.as_zigzag());
-        let encoded = match ArrayKind::from(array) {
-            ArrayKind::Primitive(p) => zigzag_encode(p),
-            _ => unreachable!("This array kind should have been filtered out"),
-        }
-        .unwrap();
+    ) -> VortexResult<OwnedArray> {
+        let zigzag_like = like.map(|like_arr| like_arr.as_array_ref());
+        let encoded = if array.encoding().id() == Primitive::ID {
```

Comment:
Since we validate in can_compress that the array is primitive, we can just `as_primitive` and panic here.